### ISSUE TITLE
Cluster does not start if cluster_name_label is anything but "version"

### ISSUE THREAD

**Issue by jens-totemic** [zalando-incubator/postgres-operator#411]
I'm hitting a somewhat strange issue when starting up a new cluster, which gets stuck in the initialization stage.

In the example cluster configuration https://github.com/zalando-incubator/postgres-operator/blob/master/manifests/configmap.yaml#L8 the property `cluster_name_label` has the value `version`. After I changed this to the default value `cluster-name` (or any other name), my new cluster gets stuck while spinning up. 

The first postgres pod created has the label `spilo-role: replica` attached to it (it should show "master") and has the annotation: `status:  {"conn_url":"postgres://10.0.2.207:5432/postgres","api_url":"http://10.0.2.207:8008/patroni","state":"running","role":"replica","xlog_location":2`

The other postgres pods created show this annotation
`status:  {"conn_url":"postgres://10.0.3.187:5432/postgres","api_url":"http://10.0.3.187:8008/patroni","state":"stopped","role":"uninitialized"}`

Also the fist pod created shows an ongoing stream of errors of this type
```
2018-11-09 19:54:48,052 INFO: Could not take out TTL lock
2018-11-09 19:54:48,150 INFO: following new leader after trying and failing to obtain lock
2018-11-09 19:54:49,151 INFO reaped unknown pid 378
2018-11-09 19:54:54,155 INFO reaped unknown pid 506
2018-11-09 19:54:57,951 INFO: Could not take out TTL lock
2018-11-09 19:54:57,969 INFO: following new leader after trying and failing to obtain lock
2018-11-09 19:54:58,970 INFO reaped unknown pid 636
2018-11-09 19:55:03,975 INFO reaped unknown pid 764
2018-11-09 19:55:07,962 INFO: Could not take out TTL lock
```
When I change the configuration and use `cluster_name_label: version`, the cluster starts up immediately. Is there any kind of special meaning to the label `version`?

**Comment by zerg-junior** [comment]
Confirmed on `minikube`, @erthalion pls have a look.

**Comment by Jan-M** [comment]
I believe that in total the Zalando Spilo image is quite explicit in its expectation as it was ment for us and served primarily as an example of a Docker Image with Patroni and Postgres, which we often refer to.

We will have a look at the obvious config options and how this could work or be documented.

At some point it may just be beneficial to come up with a custom Docker Image for your infrastructure.

**Comment by alexeyklyukin** [comment]
This is a known issue.

The change here https://github.com/zalando/spilo/pull/230/files provides environment variables to pass custom role, scope and other labels to Spilo. What's left is for the operator is actually pass it based on the configuration options.

**Comment by aknuds1** [comment]
I think I fixed this in #437.

**Comment by erthalion** [comment]
Should be fixed now with https://github.com/zalando-incubator/postgres-operator/commit/f7058c754d69127b3ce6cc45c301ac2c59cc6f97, @jens-totemic please check.
